### PR TITLE
Interactive RigidObject python tutorial page

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Added a much needed tutorial for rigid object interactivity through the python Simulator API.

## How Has This Been Tested

New tutorial code is run in `tests/test_examples.py`.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dhruvbatra** [comment]
I am unable to build documentation: 

```bash
dbatra at dbatra-mbp in ~/projects/habitat-sim/docs on rigid-object-tutorial
$ ./build-public.sh
...
...
Traceback (most recent call last):
  File "./m.css/documentation/python.py", line 2636, in <module>
    module = SourceFileLoader(name, args.conf).load_module()
  File "<frozen importlib._bootstrap_external>", line 399, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 823, in load_module
  File "<frozen importlib._bootstrap_external>", line 682, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "conf-public.py", line 9, in <module>
    from conf import *
  File "/Users/dbatra/projects/habitat-sim/docs/conf.py", line 17, in <module>
    import habitat_sim  # NOQA
  File "/Users/dbatra/projects/habitat-sim/docs/../habitat_sim/__init__.py", line 16, in <module>
    from habitat_sim.simulator import *
  File "/Users/dbatra/projects/habitat-sim/docs/../habitat_sim/simulator.py", line 21, in <module>
    from habitat_sim.physics import MotionType
  File "/Users/dbatra/projects/habitat-sim/docs/../habitat_sim/physics.py", line 5, in <module>
    from habitat_sim._ext.habitat_sim_bindings import MotionType, VelocityControl
ImportError: cannot import name 'VelocityControl'
```

File: docs/pages/rigid-object-tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+.. contents::
+    :class: m-block m-default
+
+First, download the `example objects`_ and extract them into data/objects/.
+
+The example code below is runnable via:
+
+.. code:: shell-session
+
+    $ python examples/tutorials/rigid_object_tutorial.py
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:21]:
```suggestion
    $ python /path/to/habitat-sim/examples/tutorials/rigid_object_tutorial.py
```

Because otherwise we're assuming where the user is calling this from.

File: docs/pages/rigid-object-tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+Basic rigid body simulation can be achieved by simply loading a template, instancing an object, and stepping the physical world.
+In this example, a sphere object template is loaded and the object is instanced in the scene above the table.
+When the simulation is stepped, it falls under the force of gravity and reacts to collisions with the scene.
+
+.. include:: ../../examples/tutorials/rigid_object_tutorial.py
+    :code: py
+    :start-after: # [basics]
+    :end-before: # [/basics]
+
+.. image:: images/rigid-object-tutorial-images/sim_basics.gif
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:49]: Gif looks very cool. But the video at my end doesn't have the sphere moving. Likely the same issue at #613. 

sim_basics2

File: docs/pages/rigid-object-tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+A sphere is then thrown at the boxes by applying an initial velocity.
+
+Note that forces and torques are treated as constant within each call to :ref:`Simulator.step_physics` and are cleared afterward.
+
+.. include:: ../../examples/tutorials/rigid_object_tutorial.py
+    :code: py
+    :start-after: # [dynamic_control]
+    :end-before: # [/dynamic_control]
+
+.. image:: images/rigid-object-tutorial-images/dynamic_control.gif
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:66]: Same problem. Object stuck in initial position, ball not thrown.

File: docs/pages/rigid-object-tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+In the example below, a kinematic can is placed in the scene which will not react to physical events such as collision with dynamically simulated objects.
+However, it will still act as a collision object for other scene objects as in the following example.
+
+.. include:: ../../examples/tutorials/rigid_object_tutorial.py
+    :code: py
+    :start-after: # [kinematic_interactions]
+    :end-before: # [/kinematic_interactions]
+
+.. image:: images/rigid-object-tutorial-images/kinematic_interactions.gif
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:85]: Same problem as above. Cans don't fall.

File: docs/pages/rigid-object-tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+To move a kinematic object, the state can be set directly before each simulation step.
+This is useful for synchronizing the simulation state of objects to a known state such as a dataset trajectory, input device, or motion capture.
+
+.. include:: ../../examples/tutorials/rigid_object_tutorial.py
+    :code: py
+    :start-after: # [kinematic_update]
+    :end-before: # [/kinematic_update]
+
+.. image:: images/rigid-object-tutorial-images/kinematic_update.gif
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:101]: 1. These videos work at my end. 

2. I really like these examples, but can we show the LoCoBot mesh moving on the floor? Like these https://github.com/facebookresearch/habitat-sim/pull/486

I suspect a lot of people will come looking to see how to control robots. This might also be a good place to talk about attaching a scene node to an instanced physics object so people can record videos from attached/overhead camera as shown here: #492.

File: docs/pages/rigid-object-tutorial.rst (hunk 2 above)
Comment (aclegg3) [reply docs/pages/rigid-object-tutorial.rst:49]: @dhruvbatra Did you install Bullet and compile with `--bullet`? If not there will not be any dynamics. You should see an indication in the console output that bullet was not available.

Comment (dhruvbatra) [reply docs/pages/rigid-object-tutorial.rst:49]: Ah, that might be it. Let me try again.

Comment (dhruvbatra) [reply docs/pages/rigid-object-tutorial.rst:49]: Following up. I am rebuilding with bullet but still seeing errors like these: 

```bash
(habitat)
dbatra at dbatra-mbp in ~/projects/habitat-sim on rigid-object-tutorial [?]
$ rm -rf build/
(habitat)
dbatra at dbatra-mbp in ~/projects/habitat-sim on rigid-object-tutorial [?]
$ python setup.py install --bullet
...
<successful>
...
dbatra at dbatra-mbp in ~/projects/habitat-sim on rigid-object-tutorial [?]
$ python examples/tutorials/rigid_object_tutorial.py
...
---
Physics was enabled and Bullet physics engine was specified, but the project is built without Bullet support. Objects added to the scene will be restricted to kinematic updates only. Reinstall with --bullet to enable Bullet dynamics.
---
```

I don't believe this error message because the viewer works: 
```bash
(habitat)
dbatra at dbatra-mbp in ~/projects/habitat-sim on rigid-object-tutorial [?]
$ ./build/viewer --enable-physics data/scene_datasets/habitat-test-scenes/van-gogh-room.glb
```

I am able to add object with "o". And I see them colliding.

Comment (aclegg3) [reply docs/pages/rigid-object-tutorial.rst:49]: This sounds like it could be an issue with python module building. @erikwijmans does this seem possible?

Comment (dhruvbatra) [reply docs/pages/rigid-object-tutorial.rst:49]: Just following up on this -- I'm not sure what changed, but after a clean clone and build, this example is working for me, so we're ready to go. 

One minor thing -- I'm seeing a slightly different collision in `dynamic_control.mp4` (attached) than what you uploaded. I suspect it's something to do with system-specific initial conditions? 

dynamic_control

File: docs/pages/rigid-object-tutorial.rst (hunk 5 above)
Comment (aclegg3) [reply docs/pages/rigid-object-tutorial.rst:102]: Thanks for this suggestion, I've added a new section to the tutorial demonstrating this.

File: docs/pages/rigid-object-tutorial.rst (hunk 2 above)
Comment (aclegg3) [reply docs/pages/rigid-object-tutorial.rst:50]: It looks like your box object's COM is off. I modified this somewhat recently. Be sure to download the updated objects and replace them in the data/objects folder. (same link).

The gifs were also out of date, I've updated them for the new version. Try it now 😄

Comment (dhruvbatra) [reply docs/pages/rigid-object-tutorial.rst:50]: Yes, can confirm, it was the assets. After pulling to latest on this branch:

with my old objects: 
dynamic_control

with new objects:
dynamic_control2

Resolved.

File: .circleci/config.yml [hunk 6]
```diff
@@ -298,6 +298,9 @@ jobs:
                 wget http://dl.fbaipublicfiles.com/habitat/objects_v0.1.zip
                 unzip objects_v0.1.zip -d data/objects/
                 rm objects_v0.1.zip
+                wget http://dl.fbaipublicfiles.com/habitat/locobot_merged.zip
+                unzip locobot_merged.zip -d data/objects/
+                rm locobot_merged.zip
```
Comment (dhruvbatra) [inline .circleci/config.yml:303]: I understand the workflow now after reading this. I had initially unzipped locobot_merged into the data/locobot_merged folder and was surprised when the example script crashed. (I do now see that in your instructions, you do ask them to be unzipped into data/objects/)

Instead of dumping everything into a single objects folder, would it be better to unzip locobot_merged.zip into data/locobot_merged so that way each asset-dataset is contained in it's own folder?

File: docs/pages/rigid-object-tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+In this example, the agent is embodied by a rigid robot asset and the :ref:`habitat_sim.physics.VelocityControl` structure is used to control the robot's actions.
+
+.. include:: ../../examples/tutorials/rigid_object_tutorial.py
+    :code: py
+    :start-after: # [embodied_agent]
+    :end-before: # [/embodied_agent]
+
+.. image:: images/rigid-object-tutorial-images/robot_control.gif
+    :width: 20em
```
Comment (dhruvbatra) [inline docs/pages/rigid-object-tutorial.rst:147]: This video was soo pretty!! This is awesome!

One largely minor request, which can be ignored because it's covered in other tutorials -- can we also show the depth frame (so make the sensor rgba and depth)? But as I said, it's not critical, so feel free to ship without it.

**Comment by dhruvbatra** [comment]
Reviewing this PR was like unwrapping a basket of gifts. So many goodies.

File: .circleci/config.yml (hunk 6 above)
Comment (aclegg3) [reply .circleci/config.yml:303]: Hm, in this case it may be best to add another directory like we have for scenes (`data/scene_datasets/`) and do `data/object_datasets/objects/` and `data/object_datasets/locobot_merged/`.

File: docs/pages/rigid-object-tutorial.rst (hunk 7 above)
Comment (aclegg3) [reply docs/pages/rigid-object-tutorial.rst:147]: No problem, easy addition. 👍

File: .circleci/config.yml (hunk 6 above)
Comment (aclegg3) [reply .circleci/config.yml:303]: Changing this would break some existing user workflows I think. Let's talk about data organization and make a separate PR to update things to clarify the breaking nature of this change.

Comment (dhruvbatra) [reply .circleci/config.yml:303]: Sounds good.
